packages/test/test-end-to-end-tests/src/test/gc/gcTombstone.spec.ts

```diff
@@ -84,472 +62,732 @@ describeNoCompat("GC DataStore Tombstoned When It Is Sweep Ready", (getTestObjec
-        const dataObject = await requestFluidObject<ITestDataObject>(summarizingContainer, unreferencedId);
-
-        // Wait enough time so that the datastore is sweep ready
-        await delay(remainingTimeUntilSweepMs);
-
-        await sendOpToUpdateSummaryTimestampToNow(summarizingContainer);
-
-        // The datastore should be tombstoned now
-        await summarize(summarizer);
-
-        // Note: if a user makes a request that looks like this, we will also think that the request is via handle
-        const request: IRequest = { url: unreferencedId, headers: { [RuntimeHeaders.viaHandle]: true } };
-        const response = await dataObject._context.IFluidHandleContext.resolveHandle(request);
-
-        assert(response !== undefined, `Expecting a response!`);
-        assert(response.status === 404);
-        assert(response.value.startsWith(`Datastore removed by gc`));
-    });
+        it("adds tombstone data stores information to tombstone blob in summary", async () => {
+            const mainContainer = await provider.makeTestContainer(testContainerConfig);
+            const mainDataStore = await requestFluidObject<ITestDataObject>(mainContainer, "default");
+            const mainDataStoreUrl = `/${mainDataStore._context.id}`;
+            await waitForContainerConnection(mainContainer);
+
+            const summarizer = await createSummarizer(
+                provider,
+                mainContainer,
+                undefined /* summaryVersion */,
+                gcOptions,
+                mockConfigProvider(settings),
+            );
+
+            // Create couple of data stores.
+            const newDataStore = await requestFluidObject<ITestDataObject>(
+                await mainDataStore._context.containerRuntime.createDataStore(TestDataObjectType), "");
+            const newDataStore2 = await requestFluidObject<ITestDataObject>(
+                await mainDataStore._context.containerRuntime.createDataStore(TestDataObjectType), "");
+            const newDataStoreUrl = `/${newDataStore._context.id}`;
+            const newDataStore2Url = `/${newDataStore2._context.id}`;
+
+            // Add the data stores' handle so that they are live and referenced.
+            mainDataStore._root.set("newDataStore", newDataStore.handle);
+            mainDataStore._root.set("newDataStore2", newDataStore2.handle);
+
+            // Remove the data stores' handle to make them unreferenced.
+            mainDataStore._root.delete("newDataStore");
+            mainDataStore._root.delete("newDataStore2");
+
+            // Summarize so that the above data stores are marked unreferenced.
+            await provider.ensureSynchronized();
+            const summary = await summarizeNow(summarizer);
+            validateTombstoneState(summary.summaryTree, undefined /* tombstones */, []);
+
+            // Wait for sweep timeout so that the data stores are tombstoned.
+            await delay(sweepTimeoutMs + 10);
```

Comment:
The reason there was no need to add any buffer, was because there is time between container creation and when we actually need to delay, that already adds enough buffer.

Reply:
I see. Although the unreference happens during summarize and not when the unreferenced op is sent right (well it uses the timestamp of the unreferenced op during summarize).
Adding the delay makes it determinstic rather on relying on something explicit.